File: BeeCrowd Python/Iniciante/1847.py
def verificar_figura(A,B,C):
    if B < A and C >= B:
        print(":)")
    elif B > A and C <= B:
        print(":(")
    elif (B > A and C > B) and (abs(B-C) < abs(A-B)):
        print(":(")
    elif B > A and C > B and (abs(B-C) >= abs(A-B)):
        print(":)")
    elif B < A and C < B and (abs(B-C) < abs(A-B)):
        print(":)")
    elif (B < A and C < B) and (abs(B-C) >= abs(A-B)):
        print(":(")
    elif A == B:
        if C > B:
            print(":)")
        else:
            print(":(")
         
def main():
    A,B,C = map(int,input().split())

    verificar_figura(A,B,C)

main()

File: BeeCrowd C++/Iniciante/1847.cpp
#include <iostream>
using namespace std;

string verificar_figura(int A,int B, int C){
    if (B < A and C >= B){
        return ":)";
    }else if (B > A and C <= B){
        return ":(";
    }
    else if ((B > A and C > B) and (abs(B-C) < abs(A-B))){
        return ":(";
    }
    else if (B > A and C > B and (abs(B-C) >= abs(A-B))){
        return ":)";
    }
    else if (B < A and C < B and (abs(B-C) < abs(A-B))){
        return ":)";
    }
    else if ((B < A and C < B) and (abs(B-C) >= abs(A-B))){
        return ":(";
    }
    else if (A == B){
        if (C>B){
            return ":)";
        }else{
            return ":(";
        }
    }
    return 0;
}

int main(){
    int A,B,C;
    
    cin >> A >> B >> C;
    
    cout << verificar_figura(A,B,C) << endl;
    return 0;
}
